### PR TITLE
Add optimized version of compute orientorder/atom from Tomas Oppelstr…

### PR DESCRIPTION
**Pull request by stanmoore1**
…up (LLNL)

**Summary**

Add optimized version of compute_orientorder_atom from Tomas Oppelstrup (LLNL):
- reduces the number of iterations in the L,m1,m2 loops (by a factor of up to 10x)
- avoids evaluation of Ylm functions of negative m

**Related Issue(s)**

None

**Author(s)**

Tomas Oppelstrup (LLNL), changes ported to Kokkos by Stan Moore (SNL)

**Licensing**

By submitting this pull request, I agree, that my contribution will be included in LAMMPS and redistributed under either the GNU General Public License version 2 (GPL v2) or the GNU Lesser General Public License version 2.1 (LGPL v2.1).

**Backward Compatibility**

Yes

### CODE REVIEW

**Comment by stanmoore1** [comment]
This passes my tests.

**Comment by athomps** [comment]
I am looking at it now.

**Comment by athomps** [comment]
Why does c_ change to c2_ in the thermo output header. Is that a new feature?

**Comment by stanmoore1** [comment]
@akohlmey I fixed merge conflicts but is there some way to easily reapply the formatting changes?

**Comment by stanmoore1** [comment]
I did some of the reformatting by hand but eventually gave up since it was too much...

**Comment by akohlmey** [comment]
> @akohlmey I fixed merge conflicts but is there some way to easily reapply the formatting changes?

Can you show me which changes?

File: src/compute_orientorder_atom.cpp [hunk 1]
```diff
@@ -621,68 +637,92 @@ double ComputeOrientOrderAtom::associated_legendre(int l, int m, double x)
...
+  struct {
+    double operator() (const int a, const int b, const int c,
+                       const int alpha, const int beta,const int gamma,
+                       const int t) {
+      return factorial(t)*factorial(c-b+t+alpha)*factorial(c-a+t-beta) * factorial(a+b-c-t)*factorial(a-t-alpha)*factorial(b-t+beta);
     }
+  } x;
+  const double
+    sgn = 1 - 2*((a-b-gamma)&1),
+    g = sqrt(triangle_coeff(lmax,lmax,lmax)) * sqrt(factorial(a+alpha)*factorial(a-alpha)*
```
Comment (stanmoore1) [inline src/compute_orientorder_atom.cpp:712]: @akohlmey this new code isn't formatted--i.e. changing`(a+alpha)*factorial(a-alpha)` to `(a + alpha) * factorial(a - alpha)`

Comment (akohlmey) [reply src/compute_orientorder_atom.cpp:712]: `clang-format -i <filename1> <filename2> ...` should do the trick. I can do it for you, if needed.

Comment (stanmoore1) [reply src/compute_orientorder_atom.cpp:712]: @akohlmey it would be great if you could clang-format this, thanks

**Comment by stanmoore1** [comment]
This now passes in Jenkins. I think the previous failure was bogus, ready for merge.

**Pull request merged**